libnmstate/nm/dns.py

```diff
@@ -106,3 +107,16 @@ def add_dns(setting_ip, dns_state):
         setting_ip.add_dns(server)
     for search in dns_state.get(DNS.SEARCH, []):
         setting_ip.add_dns_search(search)
+
+
+def get_dns_config_iface_names(acs_and_ipv4_profiles, acs_and_ipv6_profiles):
+    """
+    Return a list of interface name which holds the static DNS configurations.
+    """
+    iface_names = []
+    client = nmclient.client()
+    for ac, ip_profile in chain(acs_and_ipv6_profiles, acs_and_ipv4_profiles):
+        if not ip_profile.props.dns and not ip_profile.props.dns_search:
+            continue
```

Comment:
It will be simpler to understand this in a more direct form:
```
if ip_profile.props.dns or ip_profile.props.dns_search:
    iface_names.append(...)
```